src/integrators/implicit_integrators.jl

```diff
@@ -35,17 +35,17 @@ end
   @pack integrator = t,dt,u
 end
 
-type RHS_IE{F,uType,tType,DiffCacheType,SizeType,uidxType} <: Function
+type RHS_IE{F,uType,tType,DiffEqDiffTools.DiffCacheType,SizeType,uidxType} <: Function
   f::F
   u_old::uType
   t::tType
   dt::tType
-  dual_cache::DiffCacheType
+  dual_cache::DiffEqDiffTools.DiffCacheType
```

Comment:
No, this is a parametric type, not the abstract type. Don't put this in the DiffTools module. Revert this.